lib/rexml/xpath_parser.rb

```diff
@@ -78,7 +78,19 @@ def variables=( vars={} )
 
     def parse path, node
       path_stack = @parser.parse( path )
-      match( path_stack, node )
+      if node.is_a?(Array)
```

Comment:
Can we remove this check because we dropped support for `Array` here?

Reply:
@kou 

If this check is remove, the following deprecated message will no longer be output...

https://github.com/ruby/rexml/blob/cd575a10cac58eb47f235ed186060ac65ffb5284/lib/rexml/xpath_parser.rb#L139-L144

Reply:
Can we avoid duplicated `Array` checks?

Reply:
I added the deprecation check to `xpath_parser.rb`, but maybe I should have added the deprecation directly to `xpath.rb`'s `def XPath::first`, `def XPath::each` and `def XPath::match`.

(To do that, we need to be careful about the return value especially for `XPath.each(e,p) #=> Enumerator` and `XPath.each(e,p){} #=> Array`)